### PR TITLE
log: Add PrometheusCounterWrapper

### PR DESCRIPTION
**Pull request by fishy**
This is the prometheus version of metricsbp.LogWrapper.

### CODE REVIEW

File: log/wrapper.go [hunk 1]
```diff
@@ -272,6 +273,27 @@ func ErrorWithSentryWrapper() Wrapper {
...
+// Please note that it's not possible to deserialize this Wrapper directly from
+// yaml, so you usually need to override it in your main function, after
+// baseplate.ParseConfigYAML call, for example:
+//
+//     if err := baseplate.ParseConfigYAML(&cfg); err != nil {
+//       log.Fatal(err)
+//     }
+//     cfg.Config.Tracing.Logger = log.PrometheusCounterWrapper(
+//       cfg.Config.Tracing.Logger,       // delegate
+//       tracingFailureCounter.With(nil), // counter
```
Comment (pacejackson) [inline log/wrapper.go:288]: why the `.With(nil)`?

**Review by pacejackson — APPROVED** [review]

File: log/wrapper.go (hunk 1 above)
Comment (fishy) [reply log/wrapper.go:288]: In a typical prometheus counter, the global variable you defined is a [`CounterVec`](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#CounterVec), and [`With`](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#CounterVec.With) gives you an actual [`Counter`](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#Counter).

The args to `With` is the label values. For things used globally as a log wrapper, you usually don't really have any labels defined for the counter, thus the `nil`.

Comment (pacejackson) [reply log/wrapper.go:288]: would it make sense to just do `counter = counter.With(nil)` in PrometheusCounterWrapper before returning the function?

Comment (fishy) [reply log/wrapper.go:288]: actually let me expand the example a little bit.

**Review by pacejackson — APPROVED** [review]

File: log/wrapper.go [hunk 2]
```diff
@@ -272,6 +273,36 @@ func ErrorWithSentryWrapper() Wrapper {
...
+// yaml, so you usually need to override it in your main function, after
+// baseplate.ParseConfigYAML call, for example:
+//
+//     // a global variable
+//     var tracingFailures = promauto.NewCounterVec(prometheus.CounterOpts{
+//       Namespace: "myservice",
+//       Subsystem: "tracing",
+//       Name:      "failures_total",
+//       Help:      "Total number of failures when sending tracing spans to the sidecar",
+//     }, nil /* no labels */).With(nil /* no labels */)
```
Comment (kylelemons) [inline log/wrapper.go:289]: Don't use a Vec if you don't have labels.

If you're going to have an example with labels, the With call should happen at the PrometheusCounterWrapper call-site.

**Review by kylelemons — CHANGES_REQUESTED** [review]

**Review by kylelemons — APPROVED** [review]

**Review by konradreiche — APPROVED** [review]

**Pull request merged**